### PR TITLE
Add room topic to room info

### PR DESCRIPTION
**Pull request by Arnei**
For very long room titles and/or windows with small width/resolution, the room topic in the room header becomes hidden.
Bildschirmfoto vom 2022-09-30 09-12-08

To make it easily accessible, I suggest adding the room topic to the room information.
Bildschirmfoto vom 2022-09-30 09-12-29

Signed-off-by: Arne Wilken arnepokemon@yahoo.de

---
Here's what your changelog entry will look like:

## ✨ Features
 * Add room topic to room info ([\#9334](https://github.com/matrix-org/matrix-react-sdk/pull/9334)). Contributed by @Arnei.

### CODE REVIEW

**Review by gsouquet — CHANGES_REQUESTED** [review]
Thank you for raising this pull request!

I'm not entirely convinced that having this information in the `RoomSummaryCard` is the way to go. I would try to fiddle with the CSS to make sure that the topic is always a tiny bit visible in the room header.
That way we can show the topic modal when clicking it that displays the expanded room title and expanded room topic. 
But I will let a designer give their review

Regarding the code you wrote, I believe we should make `InfoTopic` generic as it's too tied to the context where it's used at the moment. And for this pull request to be accepted you'll need to have some test coverage of your changes

**Comment by Arnei** [comment]
Hi gsouquet, thanks for your very fast feedback. 

I am also not 100% sure adding the room topic to `RoomSummaryCard` is the best way of doing this. I merely assumed it would be more intuitive for users to click on the "room info" button when they want to know the topic of the room.

File: src/components/views/right_panel/RoomSummaryCard.tsx [hunk 1]
```diff
@@ -244,6 +246,65 @@ const onRoomSettingsClick = (ev: ButtonEvent) => {
...
+    // Use styles to expand or shrink the component based on the toggle
+    const styleNotExpanded = {
+        lineHeight: `${lineHeight}px`,
+        webkitLineClamp: `${lines}`,
+    };
+
+    const styleExpanded = {
+        height: `${scrollHeight}px`,
+        lineHeight: `${lineHeight}px`,
+        WebkitLineClamp: 'none',
```
[the remark this answers is not in the document]
Comment (Arnei) [reply src/components/views/right_panel/RoomSummaryCard.tsx:296]: Thanks for pointing that out, should be fixed now.

File: src/components/views/right_panel/RoomSummaryCard.tsx [hunk 2]
```diff
@@ -244,6 +246,65 @@ const onRoomSettingsClick = (ev: ButtonEvent) => {
     PosthogTrackers.trackInteraction("WebRightPanelRoomInfoSettingsButton", ev);
 };
 
+const InfoTopic: React.FC<IAppsSectionProps> = ({ room }) => {
+    const topic = useTopic(room);
+    const body = topicToHtml(topic?.text, topic?.html);
+
+    const lineHeight = 16;                      // Basically font-size
```
[the remark this answers is not in the document]
Comment (Arnei) [reply src/components/views/right_panel/RoomSummaryCard.tsx]: Wasn't aware you could do that, TIL.

**Comment by florianduros** [comment]
One solution also is to display the topic name in the topic tooltip. Currently we are displaying `Room options`, maybe it'll more relevant to display here the full topic name.

**Comment by MidhunSureshR** [comment]
Hello :wave:, thanks for the PR but unfortunately this is no longer relevant with the new room header which can be enabled in the labs settings

**Pull request closed without merging**